proposals/052-cryptography-leg-1.md

```diff
@@ -0,0 +1,439 @@
+- Cryptographic "kitchen sinks"
+    - [cryptonite](https://hackage.haskell.org/package/cryptonite)
+    - [crypton](https://hackage.haskell.org/package/cryptonite) - a fork of cryptonite
+- Bindings to libsodium
+    - [NaCl](https://github.com/serokell/haskell-crypto)
+    - [saltine](https://hackage.haskell.org/package/saltine)
+    - [sel](https://hackage.haskell.org/package/sel-0.0.1.0/candidate)
+- Bindings to botan
+    - [Z-Botan](https://hackage.haskell.org/package/Z-Botan) - part of Z-Haskell, GHC8
+- Honorable mentions
+    - [Crypto](https://hackage.haskell.org/package/Crypto) - a very old sink
+    - [crypto-api](https://hackage.haskell.org/package/crypto-api) - cryptographic classes
+
+These prior efforts can be considered to be somewhat partially successful in presenting Haskell developers with access to cryptographic primitives, but present a significant development challenge and a high refactoring cost if dependencies become deprecated, which happens often. Of these libraries, a considerable amount are deprecated or archived, with `cryptonite`, `crypton`, and `saltine` being clear front-runners in terms of current usage and transitive dependencies (and `sel` being under active development). 
+
+Of these four, one is a fork of another (`crypton` and `cryptonite`), and the other two are bindings to `libsodium` which does not sufficiently capture our need for low-level cryptographic primitives in general. `crypton/ite` exposes a few highly- opinionated / specific classes for hashing, block and stream ciphers, but otherwise is focused on concrete implementations and functions. Both `crypton` and `cryptonite` rely on bindings to C implementations that have not undergone the same vetting as commonly-used C libraries; though the code is itself not suspect, the lack of provenance makes it ill-suited for recommendation in trusted environments, and maintenance of the C implementation poses burden of high technical skill. `libsodium` bindings are more trustworthy, but the limited options severely hamper interop with other systems unless they too are using `libsodium`, making them ill-suited for recommendation for some use cases.
+
+Following that, there is also `Z-Botan`, which has been apparently unmaintained for several years. Reviving it was considered, but would require a comprehensive and near-total rewrite in order to divest itself of `Z-Haskell` as a dependency. New bindings could instead draw significantly from `Z-Botan`, while presenting a much smaller maintenance and dependency surface.
+
+Furthermore, none of these libraries contain implementations of post-quantum cryptography schemes; although existing quantum computers are of insufficient capability to break commonly used algorithms such as Ed25519, this will not remain true forever, and it would be best to have quantum-resistant implementations ready for adoption long before such attacked become practical.
```

Comment:
> ...  before such attacked become ...

attacked -> attacks.

It's a nice point, and in this context it's often worth pointing out that post-quantum cryptography is necessary now because _it is sufficient to capture the encrypted traffic now!_ and then decode it later once there are powerful-enough quantum computers; i.e. data now is not safe against future quantum computers.